src/WalletRpcPlugin.ts

```diff
@@ -0,0 +1,152 @@
+import { Web3PluginBase, utils, validator } from "web3";
+import {
+  AddEthereumChainRequest,
+  GetOwnedAssetsRequest,
+  GetOwnedAssetsResult,
+  SwitchEthereumChainRequest,
+  UpdateEthereumChainRequest,
+  WatchAssetRequest,
+} from "./types";
+import { parseToGetOwnedAssetsResult } from "./utils";
+
+type WalletRpcApi = {
+  wallet_addEthereumChain: (param: AddEthereumChainRequest) => void;
+  wallet_updateEthereumChain: (param: UpdateEthereumChainRequest) => void;
+  wallet_switchEthereumChain: (param: SwitchEthereumChainRequest) => void;
+  wallet_getOwnedAssets: (param: GetOwnedAssetsRequest) => GetOwnedAssetsResult;
+  wallet_watchAsset: (param: WatchAssetRequest) => boolean;
+};
+
+/**
+ * This Web3.js plugin adds support for various wallet-related RPC methods.
+ *
+ * @example
+ * Initialize the plugin
+ *
+ * ```typescript
+ * import { Web3 } from "web3";
+ * import { WalletRpcPlugin } from "web3-plugin-wallet-rpc";
+ *
+ * const web3 = new Web3("https://eth.llamarpc.com");
+ * web3.registerPlugin(new WalletRpcPlugin());
+ * ```
+ */
+export class WalletRpcPlugin extends Web3PluginBase<WalletRpcApi> {
+  public pluginNamespace = "walletRpc";
+
+  public constructor() {
+    super();
+  }
+
+  /**
+   * Request to add a new chain to the user's wallet.
+   *
+   * See [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085) for more details.
+   *
+   * @param param - See {@link AddEthereumChainRequest}
+   * @returns a Promise that resolves if the request is successful
+   */
+  public async addEthereumChain(param: AddEthereumChainRequest): Promise<void> {
```

Comment:
wdyt about removing `Ethereum` from the function names?
The usage would look like:

```ts
web3.walletRpc.addChain();
web3.walletRpc.switchChain();
web3.walletRpc.updateChain();
```
etc.

Reply:
I can see arguments for both approaches, but I think it's best to maintain consistency with the RPC method names.